src/app/file-manager/file-download-modal/file-download-modal.component.ts

```diff
@@ -74,15 +188,18 @@ export class FileDownloadModalComponent  {
   }
 
   setItems() {
-    this.data.selectedFiles.filter(file => file.type === 'File')
+    this.selectedFiles.filter(file => file.type === 'File')
       .forEach(file => {
-        const dataFile: Item = new Item(file.id, file.name, file.updated, file.created,
-          file.size, file.mediaLink, file.path, this.directory,
-          Type.DOWNLOAD, ItemStatus.PENDING, '', '', '', this.preserveStructure, false, '');
-        this.downloadFiles.push(dataFile);
-        this.store.dispatch(new Transferables.AddItem(dataFile));
-        this.done.emit(true);
-        this.router.navigate(['/status']);
+        if (!this.filesMap.has(file.id)) {
+          this.filesMap.set(file.id, file);
+          const dataFile: Item = new Item(file.id, file.name, file.updated, file.created,
+            file.size, file.mediaLink, file.path, this.directory,
+            Type.DOWNLOAD, ItemStatus.PENDING, '', '', '', this.preserveStructure, false, '', file.displayName, '');
```

Comment:
Why do we store `preserveStructure` for each item. Isn't the value the same for all of the items?

Reply:
Items could  be of different types : Type.DOWNLOAD, Type.UPLOAD, ??, and each type / batch could have a different preserveStructure value. That is the original reason to have it this way. 

We may think another approach but doesn't seem to be a trivial change to be addressed in a PR. We could create a story for that if it is really worth it.